test/rbi/rewriters/annotate_test.rb

```diff
@@ -0,0 +1,114 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module RBI
+  class AnnotateTest < Minitest::Test
+    def test_add_annotation_to_nodes
+      rbi = RBI::Parser.parse_string(<<~RBI)
+        module A
+          FOO = type_member
+
+          class B
+            attr_reader :a
+
+            def m1; end
+            def self.m2; end
+          end
+        end
+
+        class C < T::Struct
+          const :a, String
+          prop :b, String
+        end
+      RBI
+
+      rbi.annotate!("test")
```

Comment:
Here, invoking annotate adds the comment annotation to all members of the RBI. I find that a bit unexpected, I would have thought it would only annotate the root level of the RBI and that we'd need to invoke `annotate!` on the specific nodes.

Is there a reason for this design?

Reply:
Hmmm good point.

The idea is to provide a semi-automatic way to resolve merge conflict against manually create definitions inside shims.

Given this Tapioca generated RBI:
```rb
# left.rbi

class A
  def a1; end
  def a2(x); end
  def a3(x); end
end
```

And this shim:

```rb
# right.rbi

class A
  def a1; end

  def a2(y); end

  @shim
  def a3(x, y); end
end
```

Running `rbi merge --keep-annotated=shim` should give you this:

```rb
# merged.rbi

class A
  def a1; end

  <<<<<<< left.rbi
  def a2(x); end
  =======
  def a2(y); end
  >>>>>>> right.rbi

  @shim
  def a3(x, y); end
end
```

The current PR aims at providing an automated way to create those `@shim` annotations.

Maybe the annotate rewriter should take some options like `annotate_scopes`, `annotate_properties`, `annotate_top_level_only` to control the behaviour? wdyt?